### DISCUSSION TITLE
Can micropython for stm32f407 allow firmware to be placed on spi flash?

### DISCUSSION THREAD

**Discussion by lbuque** [micropython/micropython#15235] [STM32 / Pyboard] [answered]
The internal flash capacity of stm32f407 is too small.

**Comment by davefes** [comment]
Might be some clues here:
https://github.com/nspsck/STM32F411CEU6_BlackPill_Micropython
in `Storage Modification`

**Comment by andrewleech** [comment, marked as answer]
On stm32 firmware can only execute directly from SPI flash that works in "memory mapped" mode. 
Looking at "AN4760 Quad-SPI interface on STM32 microcontrollers and microprocessors" it doesn't look like 407 supports QSPI and I don't think any of the "normal" SPI interfaces support memory mapped mode.

However you can certainly get any stm set up so the filesystem is stored on external flash, LFS or Fat. This can be used for your python / mpy-cross'd mpy files perfectly well, just not frozen code compiled into a firmware image

**Reply by lbuque** [reply]
you are right.
